Cinema system/Cinema system/Services/AuthenticationService.cs

```diff
@@ -39,23 +39,37 @@ public bool CheckUserNameUsed(string userName)
 
         public bool CheckCorrectEmail(string email)
         {
-            return Regex.IsMatch(email,
-                @"^(?("")("".+?(?<!\\)""@)|(([0-9a-z]((\.(?!\.))|[-!#\$%&'\*\+/=\?\^`\{\}\|~\w])*)(?<=[0-9a-z])@))" +
-                @"(?(\[)(\[(\d{1,3}\.){3}\d{1,3}\])|(([0-9a-z][-0-9a-z]*[0-9a-z]*\.)+[a-z0-9][\-a-z0-9]{0,22}[a-z0-9]))$",
-                RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(250));
+            try
+            {
+                MailAddress m = new MailAddress(email);
+                return true;
+            }
+            catch (FormatException)
+            {
+                return false;
+            }
+        }
+
+        private string EncryptPassword (string password)
+        {
+            using var sha256 = SHA256.Create();
+
+            var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
+
+            var hash = BitConverter.ToString(hashedBytes).ToLower();
```

Comment:
Why using SHA256? Why using hash in general? Are there any alternatives? What is considered as best practice?

Reply:
Hashing is used because encryption must be one-way, i.e. upon registration, the user's password is hashed and stored in the database; after the user enters the password, the hash code will be calculated and compared with that stored in the database.
Now I read about Hash passwords in ASP.NET Core (https://docs.microsoft.com/en-in/aspnet/core/security/data-protection/consumer-apis/password-hashing?view=aspnetcore-3.1), but here, as I understand it, i need to store salt for each password too. I don't know if this is best practice. Also i read about SHA256 and SHA512.